libherokubuildpack/src/buildpack_output/mod.rs

```diff
@@ -35,6 +35,8 @@ use crate::buildpack_output::util::{prefix_first_rest_lines, prefix_lines, Parag
 use crate::write::line_mapped;
```

Comment:
I'm ready for feedback on if this is a viable path forward or what else might be needed.

## Exploring our options for solving this problem

- Give the end user a struct like `LockedWriter` that allows them to clone and manage their own IO faux duplication
  - Technically possible and quite explicit, but is gross for the buildpack maintainer. I feel we should take on this complexity, not our end users.
- Introduce a Command output function variant that only takes a single writer
  - I tried this, the function would then have to return the writer and any Result object, likely via a tuple, which is ergonomically weird.
- Introduce a non-consuming function that yields two `Write` structs that are then proxied to the actual Write output (This PR).
  - This PR: Solves the problem. The downside is that it's complex, but it's hidden from the user. There may be an easier way to implement this logic. I feel this closure type of build output streaming is the correct interface as it allows natively interfacing with the result and not having to juggle returning a build output struct.

Reply:
fwiw, i've been pretty happy with the closure variant for streaming command outputs.